File: 217-ContainsDuplicate/217-ContainsDuplicate.py
# Last updated: 2/11/2026, 11:08:29 AM
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """

        st = set()
        for num in nums:
            if num in st:
                return True
            st.add(num)
        
        return False
        

File: 217-ContainsDuplicate/217-ContainsDuplicate.cpp
// Last updated: 2/11/2026, 11:08:26 AM
#include <set>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int> st;

        for (int i = 0; i< size(nums); ++i){
            if (st.find(nums[i]) != st.end()){
                return true;
            }
            else{
                st.insert(nums[i]);
            }
        }
        return false;
        
    }
};